feat(homepage): add recordings to homepage

rcmarron opened the pull request:
## Problem

Users often go to recordings right away, so let's put it on the homepage

## Changes

This work is behind the feature flag `homepage-lists`, and it lays the groundwork to add all of the lists to the top of the homepage - starting with recordings. [Figma designs here](https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=7434%3A38795)

To make this work, this PR added 2 new components `EmptyMessage` and `CompactList`, both are in storybook.

Note: It's becoming obvious that the logic to open a recording should be pulled out into it's own component. I didn't include that work here to not explode this PR, but some of that code smells like copy-pasta.

https://user-images.githubusercontent.com/4813045/160049161-84966389-f228-4ccf-a0e5-b7ca024231cd.mov


👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Wrote tests for homepageLogic.

rcmarron committed:
layout

rcmarron committed:
add recording lists

rcmarron committed:
cleanup

rcmarron committed:
style tweaks

rcmarron committed:
add emptymessage to storybook

rcmarron committed:
add compact list to storybook

rcmarron committed:
add compact list to storybook

rcmarron committed:
add tests for homepage logic

clarkus reviewed (commented):
I see some small inconsistencies in the UI implementation... it's super close, but the spacing around the list should be 16px on all sides. The play icon diverges from the one I've been using recently - attached it here in case to swap. The only other thing is the text color for durations - it should be our standard black text. 

![play](https://user-images.githubusercontent.com/254612/160175116-bec0c34e-ae57-4172-bee5-990fb1c54b3c.svg)

We use icon color to communicate if a recording has been watched. Do you think that should be incorporated here, too? It's not super obvious, so I'd rather focus on a broader improvement, but I get that short term it makes sense to align with recordings. 

How are we determining the number of items shown here? In the designs they were capped at 5 items, but not for any good reason. Mostly curious about how the volume of new recordings is limited here.

rcmarron committed:
style tweaks

rcmarron committed:
skeleton styling

rcmarron committed:
Merge remote-tracking branch origin into homepage-lists

rcmarron committed:
update icon

rcmarron committed:
remove viewed

rcmarron committed:
fetch 5 recordings

rcmarron commented:
> I see some small inconsistencies in the UI implementation.

Good catches! I just cleaned these up. Let me know if I missed anything.

![image](https://user-images.githubusercontent.com/4813045/161635145-e10d6124-978e-4225-9045-4d50f0803717.png)


> We use icon color to communicate if a recording has been watched. Do you think that should be incorporated here, too?

It'd be awesome to do this - I'm already feeling the desire for the feature in the bit I've been using this list locally. I think it should be a separate PR though. Right now, the button in the recording list flips from blue -> muted when a user views the recording. Are you thinking we do the same here? I'd be a bit worried that a bunch of blue play buttons could be overwhelming on the homepage, but I'd lean on your judgement here.

> How are we determining the number of items shown here? In the designs they were capped at 5 items, but not for any good reason. Mostly curious about how the volume of new recordings is limited here.

Good question, I made it default to 10, and had the list scroll. I did this because I was fighting some issues with the height changing between the loading states and when the list was populated. I don't love it though - the scroll inside scroll on the homepage is annoying when you're trying to get to the dashboard, and it kind of feels like "why don't you just show me more" when you scroll down in the recording list.

I just updated it to be a fixed height list. I think it's much better. The only odd thing is that now it can be a somewhat empty list if the user only has a couple recordings.

![image](https://user-images.githubusercontent.com/4813045/161636180-331b1b8c-904a-4018-b58d-e73bf817be78.png)

rcmarron committed:
whoops

rcmarron committed:
clean up test

clarkus commented:
> It'd be awesome to do this - I'm already feeling the desire for the feature in the bit I've been using this list locally. I think it should be a separate PR though. Right now, the button in the recording list flips from blue -> muted when a user views the recording. Are you thinking we do the same here? I'd be a bit worried that a bunch of blue play buttons could be overwhelming on the homepage, but I'd lean on your judgement here.

I actually think it's a flawed pattern and don't really want to continue using it on its own. It was an existing thing with strong support, so I haven't pushed for changes yet. I think we could be way more obvious with what has been seen / not seen. 

> I just updated it to be a fixed height list. I think it's much better. The only odd thing is that now it can be a somewhat empty list if the user only has a couple recordings.

Is it just the 10 most recent recordings? Does it update based on the day or the frequency or anything? The less than 10 recordings seems like a temporary problem, but we could consider a partial empty state. Something in the last position that says "you're all caught up" or "more recordings will show here as they're available".

rcmarron commented:
> I actually think it's a flawed pattern and don't really want to continue using it on its own. It was an existing thing with strong support, so I haven't pushed for changes yet. I think we could be way more obvious with what has been seen / not seen.

Agreed, it's definitely subtle/non-obvious. I'm on board with cleaning this up as a separate project/PR.

> Is it just the 10 most recent recordings? Does it update based on the day or the frequency or anything? The less than 10 recordings seems like a temporary problem, but we could consider a partial empty state. Something in the last position that says "you're all caught up" or "more recordings will show here as they're available".

It's the most recent recordings over 1 min in duration and within the last 3 week (limited by session recording retention). In the latest rev on the PR, I also updated it to be only 5 recordings and the list never scrolls.

clarkus commented:
Cool - I think it make sense to ship the improvements and circle back with a more complete idea for recording states. We should handle that here and on the list view for recordings.

rcmarron commented:
Agreed - the other place to consider is the recordings in the persons modal. Right now, those don't have any concept of "viewed".

rcmarron committed:
Merge branch master of https://github.com/PostHog/posthog into homepage-lists

EDsCODE commented on frontend/src/scenes/project-homepage/projectHomepageLogic.tsx:
```diff
@@ -20,16 +33,61 @@ export const projectHomepageLogic = kea<projectHomepageLogicType>({
+            },
+        ],
+    }),
+
```
Does it make sense to try to reuse sessionRecordingsTableLogic here? (I'm not up to date on how that logic work) Thinking about consistency, we're subscribing to dashboardLogic here and I'm assuming you're going to need to route a bunch more logic through dashboardLogic here for other home page widgets.

(Not a blocker but could keep clarity in the logics here)

rcmarron replied:
100% agree - just moved it to using `sessionRecordingsTableLogic`. Feels much better!

rcmarron committed:
Merge branch 'master' into homepage-lists

rcmarron committed:
use recording table logic

rcmarron committed:
Merge branch master of https://github.com/PostHog/posthog into homepage-lists
